proposals/NNNN-resolve-distributed-actor-protocols.md

```diff
@@ -0,0 +1,474 @@
+  // ...
+}
+
+// 
+```
+
+The `SerializationRequirement` must be specified for all actors and protocols attempting to abstract over an actor system, because it is necessary to compile-time guarantee the correctness of values passed to such distributed actor methods. The compiler uses this associated type to verify all argument types and returned values are able to be serialized when performing remote calls, and will refuse to compile invocations would otherwise would have failed at runtime.
+
+Thanks to this new primary associated type, it is now possible to spell such our "distributed async sequence" as a generic actor, implement it once, and re-use it across any compatible actor system implementation:
+
+```swift
+distributed actor DistributedAsyncSequence<Element, ActorSystem> 
+  where Element: Sendable & Codable,
+        ActorSystem: DistributedActorSystem<any Codable> { 
+          
+  // not real implementation; simplified method to showcase introduced capabilities
+  distributed func exampleNextElement() async throws -> Element? { ... }
+}
+```
+
+Note that since the `ActorSystem` specifies a concrete `SerializationRequirement` the compiler is still able to check that all types invoked in a distributed function call conform to this protocol, i.e. we're guaranteed to be able to serialize `Element` because the ActorSystem provided must be able to handle this serialization mechanism.
+
+This also extends to distributed protocols, which are now able to abstract over an actor system, while specifying what serialization requirement they support:
+
+```swift
+protocol DistributedAsyncSequence: DistributedActor 
+    where ActorSystem: DistributedActorSystem<any Codable> {
+  associatedtype Element: Sendable & Codable
+      
+  distributed func exampleNextElement() async throws -> Element? { ... }
+}
+```
+
+Missing to specify the serialization requirement is a compile time error:
+
+```swift
+protocol DistributedAsyncSequence: DistributedActor 
+    where ActorSystem: DistributedActorSystem { 
+    // error: distributed actor protocol must specify `ActorSystem.SerializationRequirement`,
+    // you can provide it like this: DistributedActorSystem<any Codable>
+  associatedtype Element: Sendable & Codable
+      
+  distributed func exampleNextElement() async throws -> Element? { ... }
+}
+```
+
+The serialization requirement must be a `protocol` type; This was previously enforced, and remains so after this proposal. The important part is that the macro is able to synthesize an stub implementation type, that the existing resolution mechanisms can be invoked on. 
+
+## Detailed design
+
+The `@DistributedProtocol` macro generates a concrete `distributed actor` declaration as well as an extension which implements the protocol's method requirements with "stub" implementations.
+
+>  **NOTE:** The exact details of the macro synthesized code are not guaranteed to remain the same, and may change without notice.
```

Comment:
With this note here is it perhaps worth also being very explicit about what _is_ guaranteed to remain the same? I realize this is distributed throughout the proposal an implicit in Swift's source compatibility guarantee, but if we're adding this scary note I think it would be good to indicate what users _can_ actually rely on.

Reply:
Sounds good -- I think we will guarantee the existence of the `$Greeter` for a `protocol Greeter`, I'll call it out